### PR TITLE
FEATURE Rate

### PR DESCRIPTION
**Pull request by melas123**
REF #[17](https://trello.com/c/DTniFHQy)
## Details

```
* generate model
* create tests for the model
* generate controller
    * add some test
    * Fix conflicts
    * Finish the tests for rate model and controller
* Finish the tests for recipe model
```
## Others

```
* modify rate controller
```

### CODE REVIEW

File: app/controllers/recipe_controller.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@ def index
     @recipes = Recipe.all
     respond_with( @recipes )
   end
+  
+  def rate
+    @rate = Recipe.find(params[:recipe_id]).calculate_rate_for_recipe
```
Comment (SamiDghim) [inline app/controllers/recipe_controller.rb:18]: use space : 

@rate = Recipe.find( params[ :recipe_id ] ).calculate_rate_for_recipe

File: app/controllers/rates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    @rate = Rate.find(params[:id])
+  end
+
+  # GET /rates/new
+  def new
+    @rate = Rate.new
+  end
+
+  # GET /rates/1/edit
+  def edit
```
Comment (SamiDghim) [inline app/controllers/rates_controller.rb]: delete this method if you will not use it

File: app/controllers/rates_controller.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_rate
+      @rate = Rate.find(params[:id])
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
+    def rate_params
+      params[:rate]
```
Comment (SamiDghim) [inline app/controllers/rates_controller.rb]: use strong parameter instead

File: spec/controllers/rates_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    it "return global rating for a recipe " do
+      rate = FactoryGirl.create(:rate, :valid, recipe_id: recipe.id)
+      get :index, {format: :json, recipe_id: recipe.id}
+      expect(assigns(:rates)).to eq(rate)
+    end
+  end
+
+  describe "GET #show" do
+    it "assigns the requested rate as @rate" do
+      get :show, {format: :json, :id => rate.id}
```
Comment (SamiDghim) [inline spec/controllers/rates_controller_spec.rb]: plz use ruby 1.9 syntax

**Comment by moelheni** [comment]
@melas123  there's a conflit try to fix it

**Pull request merged**